cdk/kittyhawk/src/application.ts

```diff
@@ -92,31 +92,15 @@ export class Application extends Construct {
   }
 }
 
-/**
- * Helper function that first checks to make sure that the environment variable array
- * doesn't already contain the env var, then inserts it into the array.
- * @param envArray array of environment variables to insert into
- * @param envKey name of the environment variable to insert
- * @param envValue value of the environment variable
- */
-export function insertIfNotPresent(envArray: { name: string; value: string }[], envKey: string, envValue: any) {
-  const envSettingsModule = envArray.filter(env => (env.name === envKey));
-  if (envSettingsModule.length > 0) {
-    throw new Error(`${envKey} should not be redefined as an environment variable.`);
-  }
-  envArray.push({ name: envKey, value: envValue });
-
-}
-
 export class DjangoApplication extends Application {
   constructor(scope: Construct, appname: string, props: DjangoApplicationProps) {
 
-    // Have to be careful here with references when mutating things
-    let djangoExtraEnv = Array.from(props.deployment?.env || []);
-
-    // Insert DJANGO_SETTINGS_MODULE and DOMAIN
-    insertIfNotPresent(djangoExtraEnv, 'DJANGO_SETTINGS_MODULE', props.djangoSettingsModule);
-    insertIfNotPresent(djangoExtraEnv, 'DOMAIN', props.domains.map(h => h.host).join());
+    // Now, we ensure there are no duplicate env variables, even if they redefine it
+    const djangoExtraEnv = [...new Set([
```

Comment:
I think this should get rid of the mutability issue and also make sure we have only one of the envs for `DOMAIN` and `DJANGO_SETTINGS_MODULE`. 

Also is a softer way to handle users redefining these things in env rather than failing. 

Would love @eyingxuan or someone to double check the typescript logic is right

Reply:
BTW, we want to check if `DJANGO_SETTINGS_MODULE` & `DOMAIN` exist & error if it doesn’t. 

`djangoSettingsModule` is a required prop so we're fine. For React, `domain` is also required so we're fine.

The issue is `DOMAIN` for Django is an array, and it can be empty. I'm not sure if we should allow it to be an empty array. If not, the best course of action is to make sure `domains` provided is an `NonEmptyArray` type for Django. Quick win so we're good.